File: SDKs/googleapis.com/run/v1alpha1/python/src/sdk/models/shared/envfromsource.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import configmapenvsource as shared_configmapenvsource
from ..shared import secretenvsource as shared_secretenvsource


@dataclass_json
@dataclasses.dataclass
class EnvFromSource:
    r"""EnvFromSource
    Not supported by Cloud Run EnvFromSource represents the source of a set of ConfigMaps
    """
    
    config_map_ref: Optional[shared_configmapenvsource.ConfigMapEnvSource] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('configMapRef') }})
    prefix: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('prefix') }})
    secret_ref: Optional[shared_secretenvsource.SecretEnvSource] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('secretRef') }})
    

File: SDKs/googleapis.com/run/v1/java/lib/src/main/java/openapisdk/models/shared/EnvFromSource.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * EnvFromSource
 * Not supported by Cloud Run. EnvFromSource represents the source of a set of ConfigMaps
**/
public class EnvFromSource {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("configMapRef")
    public ConfigMapEnvSource configMapRef;
    public EnvFromSource withConfigMapRef(ConfigMapEnvSource configMapRef) {
        this.configMapRef = configMapRef;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("prefix")
    public String prefix;
    public EnvFromSource withPrefix(String prefix) {
        this.prefix = prefix;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("secretRef")
    public SecretEnvSource secretRef;
    public EnvFromSource withSecretRef(SecretEnvSource secretRef) {
        this.secretRef = secretRef;
        return this;
    }
}